soh/src/overlays/gamestates/ovl_file_choose/z_file_choose.c

```diff
@@ -497,10 +495,10 @@ void FileChoose_UpdateMainMenu(GameState* thisx) {
                     this->kbdY = 0;
                     this->charIndex = 0;
                     this->charBgAlpha = 0;
-                    this->newFileNameCharCount = 0;
+                    this->newFileNameCharCount = strlen(defName) >= NAME_MAX ? NAME_MAX - 1 : strlen(defName) % NAME_MAX;
```

Comment:
This ternary statement isn't necessary.  Since we've already guaranteed that players can only input at most NAME_MAX characters into the cvar, the cvar will be properly null terminated, and SetPlayerName loops through exactly NAME_MAX characters, we can just set newFileNameCharCount = strlen(defName).

Reply:
If I do that, when I put a name of 8 characters, the cursor goes to the beginning, and not on the last character..

Reply:
If you're trying to fix ImGui behavior for setting the cursor, it would be more proper to use ImGui code to do that instead of messing with the underlying game data since that character count is used elsewhere, and there could be unintended consequences for having it not exactly match the actual number of characters in the name.

Reply:
No, typing a name with 8 characters in the control has the same behavior as putting a variable and using it there with `strlen()`. But I don't need `% NAME_MAX` anymore, so I'll remove it.

Reply:
By the way, I'm referring to the File Select cursor that highlights the space after the last character typed, so that way you can delete it as it is in games that use "Link" as the default name.